src/main/msp/msp.c

```diff
@@ -209,33 +222,73 @@ static bool mspIsMspArmingEnabled(void)
     return mspArmingDisableFlags == 0;
 }
 
-#ifdef USE_SERIAL_4WAY_BLHELI_INTERFACE
-#define ESC_4WAY 0xff
+#define MSP_PASSTHROUGH_ESC_4WAY 0xff
 
-uint8_t escMode;
-uint8_t escPortIndex;
+static uint8_t mspPassthroughMode;
+static uint8_t mspPassthroughArgument;
 
 #ifdef USE_ESCSERIAL
 static void mspEscPassthroughFn(serialPort_t *serialPort)
 {
-    escEnablePassthrough(serialPort, &motorConfig()->dev, escPortIndex, escMode);
+    escEnablePassthrough(serialPort, &motorConfig()->dev, mspPassthroughArgument, mspPassthroughMode);
 }
 #endif
 
-static void mspFc4waySerialCommand(sbuf_t *dst, sbuf_t *src, mspPostProcessFnPtr *mspPostProcessFn)
+static serialPort_t *mspFindPassthroughSerialPort(void)
+{
+    serialPortUsage_t *portUsage = NULL;
+
+    switch (mspPassthroughMode) {
+    case MSP_PASSTHROUGH_SERIAL_ID:
+    {
+        portUsage = findSerialPortUsageByIdentifier(mspPassthroughArgument);
+        break;
+    }
+    case MSP_PASSTHROUGH_SERIAL_FUNCTION_ID:
```

Comment:
I actually really like this idea, this will make it so much easier for apps to 'upgrade the Tramp VTX' or suchlike. We should consider extending the approach to CLI, maybe by introducing a `serialpassthru f<function number>` syntax or similar.